misc/genstd/mapping.go

```diff
@@ -231,15 +231,95 @@ func (m *mapping) typesEqual(gnoe, goe ast.Expr) error {
 			return err
 		}
 		return nil
+	case *ast.MapType:
+		goe, ok := goe.(*ast.MapType)
+		if !ok {
+			return &mismatch
+		}
+		if err := m.typesEqual(gnoe.Key, goe.Key); err != nil {
+			return err
+		}
+		if err := m.typesEqual(gnoe.Value, goe.Value); err != nil {
+			return err
+		}
+		return nil
+	case *ast.StructType:
+		goe, ok := goe.(*ast.StructType)
+		if !ok || len(gnoe.Fields.List) != len(goe.Fields.List) {
+			return &mismatch
+		}
+		for i := 0; i < len(gnoe.Fields.List); i++ {
+			gnoField, goField := gnoe.Fields.List[i], goe.Fields.List[i]
+			if len(gnoField.Names) != len(goField.Names) {
+				return &mismatch
+			}
+			for j := 0; j < len(gnoField.Names); j++ {
+				if gnoField.Names[j].Name != goField.Names[j].Name {
+					return &mismatch
+				}
+			}
+			if err := m.typesEqual(gnoField.Type, goField.Type); err != nil {
+				return err
+			}
+		}
+		return nil
```

Comment:
What do you think about this mini cleanup?

```suggestion
		for i, gnoField := range gnoe.Fields.List {
			goField := goe.Fields.List[i]
			
			if len(gnoField.Names) != len(goField.Names) {
				return &mismatch
			}
			
			for j, gnoName := range gnofield.Names {
				if gnoName.Name != goField.Names[j].Name {
					return &mismatch
				}
			}
			
			if err := m.typesEqual(gnoField.Type, goField.Type); err != nil {
				return err
			}
		}
		return nil
```

Reply:
I'm not sure if you can move this check up, before the name loop:
```go
			if err := m.typesEqual(gnoField.Type, goField.Type); err != nil {
				return err
			}
```